### PR TITLE
ASimNetworkSession\Parsers\ASimNetworkSessionMicrosoftSysmon.yaml

### PR DESCRIPTION
**Pull request by Alekhya0824**
Required items, please complete
   
   Change(s):
   - Created a new file for WindowsEvent and added WindowsEvent parser in it. In Event file removed WindowsEvent parser and in ASimNetworkSession added WindowsEvent Parser name.

   Reason for Change(s):
   - To Split the WindowsEvent and Event parser separately

   Version Updated:
   - yes

   Testing Completed:
   - yes

   Checked that the validations are passing and have addressed any issues that are present:
   - yes

### CODE REVIEW

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please check for KQL validation failure. Thanks
image

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please respond on above. Thanks

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Validations failing please check. Thanks

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSession.yaml [hunk 1]
```diff
@@ -39,6 +39,7 @@ Parsers:
   - _ASim_NetworkSession_CiscoASA
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSession.yaml:39]: why KQL test changes?

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 Parser:
   Title: Network Session Event ASIM parser for Sysmon (Event 3)
-  Version: '0.1'
-  LastUpdated: Jan 11, 2023
+  Version: '0.2.1'
+  LastUpdated: June 3, 2024
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:4]: "LastUpdated" property exists but is not correct format. The expected format is, for example, "Jun 29, 2024"

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 3]
```diff
@@ -13,7 +13,7 @@ References:
 - Title: ASIM
   Link: https:/aka.ms/AboutASIM
 Description: |
-  This ASIM parser supports normalizing Sysmon network session events (event 3) collected using the Event or WEF (WindowsEvent table) connectors to the ASIM Process Event normalized schema. 
+  This ASIM parser supports normalizing Sysmon network session events (event 3) collected using the Event or WEF (WindowsEvent table) connectors logs ingested in 'Event' table to the ASIM Process Event normalized schema.
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:16]: please review description

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+Normalization:
+  Schema: NetworkSession
+  Version: '0.2.5'
+References:
+- Title: ASIM Network Session Schema
+  Link: https://aka.ms/ASimNetworkSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+Description: |
+  This ASIM parser supports normalizing Sysmon network session events (event 3) collected using the Event or WEF (WindowsEvent table) connectors logs ingested in 'WindowsEvent' table to the ASIM Process Event normalized schema.
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:16]: please review description

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 5]
```diff
@@ -13,7 +13,7 @@ References:
 - Title: ASIM
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:13]: ASIM schema "NetworkSession" version "0.2.5" is incorrect. The correct version for ASIM schema "NetworkSession" is "0.2.6"

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 6]
```diff
@@ -13,7 +13,7 @@ References:
...
   Link: https:/aka.ms/AboutASIM
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:14]: url correction

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 7]
```diff
@@ -1,7 +1,7 @@
 Parser:
   Title: Network Session Event ASIM parser for Sysmon (Event 3)
-  Version: '0.1'
-  LastUpdated: Jan 11, 2023
+  Version: '0.2.1'
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:3]: we don't have _Im_ parser. Need to add that too

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml [hunk 8]
```diff
@@ -0,0 +1,139 @@
+Parser:
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:1]: we don't have Im parser. Need to add that too

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+  Title: Network Session Event ASIM parser for Sysmon (Event 3)
+  Version: '0.2.1'
+  LastUpdated: June 3, 2024
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:4]: "LastUpdated" property exists but is not correct format. The expected format is, for example, "Jun 29, 2024"

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+Product:
+  Name: Windows Sysmon
+Normalization:
+  Schema: NetworkSession
+  Version: '0.2.5'
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:9]: ASIM schema "NetworkSession" version "0.2.5" is incorrect. The correct version for ASIM schema "NetworkSession" is "0.2.6"

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+References:
+- Title: ASIM Network Session Schema
+  Link: https://aka.ms/ASimNetworkSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:14]: url correction

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 12]
```diff
@@ -1,7 +1,7 @@
 Parser:
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:1]: (1) Warning: Missing recommended field [EventUid]

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml (hunk 8 above)
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:1]: (1) Warning: Missing recommended field [EventUid]

**Review by vakohl — CHANGES_REQUESTED** [review]

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml (hunk 12 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:1]: done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml (hunk 11 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:14]: done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml (hunk 10 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:9]: done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml (hunk 9 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:4]: done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml (hunk 6 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:14]: done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml (hunk 5 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:13]: done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml (hunk 4 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:16]: done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml (hunk 3 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:16]: done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml (hunk 2 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:4]: done

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
One suggestion: once you're done with the open comment change, please click on the `Resolve Conversation` button.

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSession.yaml (hunk 1 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSession.yaml:39]: updating table schema file

**Comment by vakohl** [comment]
@Alekhya0824 Please check the validation error. I think we are missing changes to imNetworkSession union parser

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml (hunk 7 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:3]: done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml (hunk 8 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmonWindowsEvent.yaml:1]: done

**Pull request closed without merging**